pkg/providers/cloudstack/cloudstack_test.go

```diff
@@ -220,6 +220,66 @@ func TestProviderGenerateCAPISpecForCreate(t *testing.T) {
 	test.AssertContentToFile(t, string(md), "testdata/expected_results_main_md.yaml")
 }
 
+func TestNamespaceMismatchBetweenClusterAndDatacenter(t *testing.T) {
```

Comment:
Why are we adding tests to the cloudstack provider? If these are supposed to cover the bug fix, can we add them in the v1alpha1 package?

Reply:
Done